node/forest_libp2p/src/service.rs

```diff
@@ -167,11 +193,39 @@ impl Libp2pService {
                         }
                         ForestBehaviourEvent::BlockSyncResponse { request_id, response, .. } => {
                             debug!("Received blocksync response (id: {:?}): {:?}", request_id, response);
-                            self.network_sender_out.send(NetworkEvent::BlockSyncResponse {
-                                request_id,
-                                response,
-                            }).await;
+                            let tx = self.bs_request_table.lock().await.remove(&request_id);
+
+                            if let Some(tx) = tx {
+                              if let Err(e) = tx.send(response) {
+                                debug!("RPCResponse receive failed: {:?}", e)
+                              }
+                            }
+                            else {
+                                debug!("RPCResponse receive failed: channel not found");
+                            };
                         }
+                        ForestBehaviourEvent::BitswapReceivedBlock(peer_id, cid, block) => {
+                            match self.db.put(&block, Blake2b256) {
+                                Ok(actual_cid) => {
+                                    trace!("saved bitswap block with returned cid {:?}, expected cid: {:?}", actual_cid, cid);
+                                }
+                                Err(e) => {
+                                    warn!("failed to save bitswap block: {:?}", e.to_string());
+                                }
+                            }
+                            self.network_sender_out.send(NetworkEvent::BitswapBlock{cid}).await;
```

Comment:
should you only send the event when the db put is successful?

Reply:
Yeah, I think thats probably what should happen, but something REALLY weird happens when i move that into the match statement... The error is too long to post here. Will show tomorrow